src/interpreter.js

```diff
@@ -155,57 +156,37 @@ const compileExpression = (expression) => {
 
 
 const compileLiteral = (literal) => {
-  if(literal.nodeType === 'NullLiteral') {
-    return `null`
+  const rules = {
+    NullLiteral: () => 'null',
+    SelfLiteral: () => 'this',
+    BooleanLiteral: ({ value }) => `${value}`,
+    NumberLiteral: ({ value }) => `${value}`,
+    StringLiteral: ({ value }) => `"${value}"`,
+    SetLiteral: ({ values }) => `new Set([ ${values.map(compileExpression).join()} ])`,
+    ListLiteral: ({ values }) => `[ ${values.map(compileExpression).join()} ]`
   }
-
-  if(literal.nodeType === 'SelfLiteral') {
-    return `this`
+  if (rules[literal.nodeType]) {
```

Comment:
Yes, sure. I did it local to compileLiteral just as a "demo case". But we can generalize it to a single object for all elements